### PR TITLE
FruitTree lumber and lumber accessories

### PR DESCRIPTION
**Pull request by IndiBowstring**
Cinnamon is in a weird spot, i'll need to figure out an elegant way of handling it later.

### CODE REVIEW

File: src/main/java/com/eerussianguy/firmalife/ClientRegisterEventsFL.java [hunk 1]
```diff
@@ -51,6 +55,8 @@ public static void registerModels(ModelRegistryEvent event)
         //Setting the model resource location for items
         for (Item i : ItemsFL.getAllEasyItems())
             ModelLoader.setCustomModelResourceLocation(i, 0, new ModelResourceLocation(i.getRegistryName().toString()));
+        for (Item i : ItemsFL.getAllFruitPoles())
```
Comment (eerussianguy) [inline src/main/java/com/eerussianguy/firmalife/ClientRegisterEventsFL.java:58]: these should just be easyItems

File: src/main/java/com/eerussianguy/firmalife/blocks/BlockFruitDoor.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.World;
+
+public class BlockFruitDoor extends BlockDoor {
+
+    private static final Map<IFruitTree, BlockFruitDoor> MAP = new HashMap<>();
+
+    public static BlockFruitDoor get(IFruitTree tree) { return MAP.get(tree); }
```
Comment (eerussianguy) [inline src/main/java/com/eerussianguy/firmalife/blocks/BlockFruitDoor.java:24]: Is this method used? We have been deliberately avoiding collecting stuff in hashmaps like this. Looks like copypasta from TFC

Comment (eerussianguy) [reply src/main/java/com/eerussianguy/firmalife/blocks/BlockFruitDoor.java:24]: If we don't need the map, we also don't have to construct fruit tree blocks using IFruitTree (and therefore we can construct a cinnamon variant easily)

File: src/main/java/com/eerussianguy/firmalife/CommonEventHandlerFL.java [hunk 3]
```diff
@@ -50,6 +57,12 @@ public static void onBlockHarvestDrops(BlockEvent.HarvestDropsEvent event)
         {
             event.getDrops().add(new ItemStack(ItemsFL.FRUIT_LEAF, 2 + Constants.RNG.nextInt(4)));
         }
+        else if (block instanceof BlockFruitTreeTrunk)
+        {
+            IFruitTree tree = ((BlockFruitTreeTrunk) block).getTree();
+            String poleName = "firmalife:" + tree.getName().toLowerCase() + "_pole";
```
Comment (eerussianguy) [inline src/main/java/com/eerussianguy/firmalife/CommonEventHandlerFL.java:63]: No, use Firmalife.MOD_ID, never use the string directly

File: src/main/resources/assets/firmalife/recipes/cherry_fence.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "XBX"
+  ],
+  "key": {
+    "X": {
+      "type": "forge:ore_dict",
+      "ore" : "poleCherry"
+    },
+    "B": {
+      "type": "forge:ore_dict",
+      "ore" : "fruitLeaf"
```
Comment (eerussianguy) [inline src/main/resources/assets/firmalife/recipes/cherry_fence.json:14]: requiring fruit leaves to build fences is rather expensive.

**Review by eerussianguy — CHANGES_REQUESTED** [review]

File: src/main/java/com/eerussianguy/firmalife/blocks/BlockFruitDoor.java (hunk 2 above)
Comment (IndiBowstring) [reply src/main/java/com/eerussianguy/firmalife/blocks/BlockFruitDoor.java:24]: I'm not 100% sure why tfc goes through the motions with the hashmaps, but it seems useful for iterating over groupings of specific items.  You're right that they aren't used right now though

File: src/main/resources/assets/firmalife/recipes/cherry_fence.json (hunk 4 above)
Comment (IndiBowstring) [reply src/main/resources/assets/firmalife/recipes/cherry_fence.json:14]: Fruit Poles don't have an intermediary step like logs > planks or planks > plank blocks, getting poles in general is quite expensive right now as a fruit tree can yield a maximum of 5 unless we bump up drop rate.  As it stands fruit wood is a rare and difficult to aquire block group

File: src/main/java/com/eerussianguy/firmalife/CommonEventHandlerFL.java [hunk 5]
```diff
@@ -50,6 +57,12 @@ public static void onBlockHarvestDrops(BlockEvent.HarvestDropsEvent event)
         {
             event.getDrops().add(new ItemStack(ItemsFL.FRUIT_LEAF, 2 + Constants.RNG.nextInt(4)));
         }
+        else if (block instanceof BlockFruitTreeTrunk)
+        {
+            IFruitTree tree = ((BlockFruitTreeTrunk) block).getTree();
+            String poleName = MOD_ID + tree.getName().toLowerCase() + "_pole";
+            event.getDrops().add(new ItemStack(ItemMisc.getByNameOrId(poleName)));
```
Comment (eerussianguy) [inline src/main/java/com/eerussianguy/firmalife/CommonEventHandlerFL.java:64]: This is bad practice but I'm going to let it slide for now. Might be possible to do BlockFruitTreeTrunk.get()? Check that

**Comment by eerussianguy** [comment]
Also need to see the cinnamon blocks make it in.

**Pull request merged**